File: Data Structures & Algorithms/search-2d-matrix/submission-0.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {

        int m=matrix.size();
        int n=matrix[0].size();
        int row=-1;
        for(int i=0;i<m;i++)
        {
            if(matrix[i][0]<=target && matrix[i][n-1]>=target)
            {
                row=i;
            }
           
        }
        if(row==-1)
        {
            return false;
        }
        int l=0;
        int r=n-1;
        while(l<=r)
        {
            int mid=l+(r-l)/2;
            if(matrix[row][mid]==target)
            {
                return true;
            }
            else if(matrix[row][mid]>target)
            {
                r=mid-1;
            }
            else
            {
                l=mid+1;
            }
        }
        return false;
        
    }
};

File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-0.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        row=[]
        col=[]
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j]==0:
                    row.append(i)
                    col.append(j)
        for r in row:
            for j in range(len(matrix[0])):
                matrix[r][j]=0
        for i in range(len(matrix)):
            for c in col:
                matrix[i][c]=0
        



        
        